flax/experimental/nnx/nnx/flaglib.py

```diff
@@ -60,7 +63,15 @@ def __call__(self, **kwargs: tp.Any):
     finally:
       FLAGS_CONTEXT.flags_stack.pop()
 
-  def get(self, name: str, default: tp.Any = None) -> tp.Optional[tp.Any]:
+  @tp.overload
+  def get(self, name: str) -> tp.Any:
+    ...
+
+  @tp.overload
+  def get(self, name: str, default: A) -> A:
```

Comment:
I think the right return type is `A | Any`, because the value stored in the context is not guaranteed to be of type `A`.